### PR TITLE
make exporters components

### PR DESCRIPTION
**Pull request by Haishi2016**
# Description

Change tracing exporters as first-class components

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [ ] Extended the documentation

Doc updates will be a separate PR after this and the main repo are updated

### CODE REVIEW

File: exporters/Readme.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  
+  Export to a string buffer. This is mostly used for testing purposes.
+
+* Zipkin
+  
+  Export to a [Zipkin](https://zipkin.io/) back-end.
+
+## Implementing a new Exporter wrapper
+
+A compliant exporter wrapper needs to implement one interfaces: `Exporter`, which has a single method:
```
Comment (yaron2) [inline exporters/Readme.md]: `one interface` instead of `one interfaces`

File: exporters/metadata.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package exporters
+
+// Metadata represents a set of exporter specific properties
+type Metadata struct {
+	Properties map[string]interface{} `json:"properties"`
+	Buffer     *string                `json:"-"`
```
Comment (yaron2) [inline exporters/metadata.go]: Why is this a pointer of a string?

File: exporters/stringexporter/string_exporter.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	Buffer *string
+}
+
+// NewStringExporter returns a new string exporter instance
+func NewStringExporter() *Exporter {
+	return &Exporter{}
+}
+
+// Exporter is an OpenCensus string exporter
+type Exporter struct {
```
Comment (yaron2) [inline exporters/stringexporter/string_exporter.go]: Can you rename this to `String`?

File: exporters/zipkin/zipkin_exporter.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	Enabled         string `json:"enabled"`
+}
+
+// NewZipkinExporter returns a new zipkin exporter instance
+func NewZipkinExporter() *Exporter {
+	return &Exporter{}
+}
+
+// Exporter is an OpenCensus zipkin exporter
+type Exporter struct {
```
Comment (yaron2) [inline exporters/zipkin/zipkin_exporter.go]: Can you rename this to `Zipkin`?

File: exporters/stringexporter/string_exporter.go (hunk 3 above)
Comment (Haishi2016) [reply exporters/stringexporter/string_exporter.go]: Why? it's under the stringexporter package

File: exporters/zipkin/zipkin_exporter.go (hunk 4 above)
Comment (Haishi2016) [reply exporters/zipkin/zipkin_exporter.go]: Why? it's under the zipkin package

File: exporters/metadata.go (hunk 2 above)
Comment (Haishi2016) [reply exporters/metadata.go]: This is to allow passing in a string buffer that is under control of the caller. This is necessary to make the component testable - see unit tests.

**Comment by yaron2** [comment]
Ready to merge?

**Pull request merged**